scripts/etch-language-tests/ml/optimiser.etch

```diff
@@ -0,0 +1,41 @@
+function main()
```

Comment:
This checks the code compiles and does not crash. It would be nice to also know if the optimisers do the right thing, if for no other reason than to protect the behaviour from being accidentally changed in the future. Add assertions, please.